packages/ai-bot/helpers.ts

```diff
@@ -350,10 +176,11 @@ export function getModifyPrompt(history: IRoomEvent[], aiBotUserId: string) {
   The user currently has given you the following data to work with:
   Cards:\n`;
   for (let card of getRelevantCards(history, aiBotUserId)) {
-    systemMessage += `Full data: ${JSON.stringify(card)}
-  You may only patch the following fields for this card: ${JSON.stringify(
-    card.attributes,
-  )}`;
+    systemMessage += `Full data: ${JSON.stringify(card)}`;
+  }
+  if (functions.length == 0) {
+    systemMessage +=
+      'You are unable to edit any cards, the user has not given you access, they need to open the card on the stack and tick "Allow access to the cards you can see at the top of your stacks" ';
```

Comment:
This seems fine for now but I wonder if it might eventually make sense to extract this string from where it appears in the UI so it can be consumed here, then there’s no danger of them falling out of sync.

Reply:
Yes, I'm hoping we can move some of this prompt construction to the host app, then keeping them in sync will be easier. Failing that we should at least get this specified in one place I agree.